src/main/java/blitz/gameoflife/RleParser.java

```diff
@@ -95,8 +84,39 @@ public int[][] decodeData(String content) {
         return grid;
     }
 
-    public int[][] loadGridFromRleFile(String filePath) throws IOException {
-        String content = readFile(filePath);
+    public int[][] getGridFromClipboard(String content) {
+        String[] lines = content.split("\n");
+
+        for (String line : lines) {
+            line = line.trim();
+            if (line.startsWith("#") || line.isEmpty()) {
+                continue;
+            }
+            if (line.startsWith("x")) {
+                parseHeader(line);
+                break;
+            }
+        }
         return decodeData(content);
     }
+
+    public String getFromClipboard() throws IOException, UnsupportedFlavorException {
+        Clipboard clipboard = Toolkit.getDefaultToolkit().getSystemClipboard();
+        String clipContent = (String) clipboard.getData(DataFlavor.stringFlavor);
+        String content;
+
+        if (clipContent.startsWith("http")) {
+            URL url = new URL(clipContent);
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+            connection.setInstanceFollowRedirects(true);
+            content = IOUtils.toString(connection.getInputStream(), "UTF-8");
+        } else if (clipContent.contains("x") && clipContent.contains("y")) {
+            content = clipContent;
+        } else if (Files.exists(Paths.get(clipContent))) {
+            content = IOUtils.toString(new FileInputStream(new File(clipContent)), "UTF-8");
+        } else {
+            throw new IOException("Clipboard content is not valid RLE data, URL, or file path.");
+        }
```

Comment:
This code belongs in your frame. The RleParser doesn't care where the file comes from.